src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -1036,7 +1036,7 @@ private boolean determineDesiredActivity()
             return false;
         }
 
-        if (getCitizenColonyHandler().getColony().isMourning() && mourning && !(citizenJobHandler.getColonyJob() instanceof JobUndertaker))
+        if (getCitizenColonyHandler().getColony().isMourning() && mourning && !(citizenJobHandler.getColonyJob() instanceof JobUndertaker) && !(citizenJobHandler.getColonyJob() instanceof AbstractJobGuard) && !(citizenJobHandler.getColonyJob() instanceof JobHealer))
```

Comment:
Wouldn't it make more sense to add a boolean function to the job API rather than adding more and more custom exceptions here?

Reply:
Agreed

Reply:
Even better is just to use an interface and smag that on jobs which need no mourning.

That interface could even specifically have a method which alles for the checking.